test/malli/transform_test.cljc

```diff
@@ -433,14 +433,86 @@
   (testing "extra-keys-transformer shouldn't break non map values"
     (is (= {:foo "bar"}
            (m/decode
-             [:map {:decode/string (fn [s] {:foo s})}
-              [:foo :string]]
-             "bar"
-             (mt/transformer
+            [:map {:decode/string (fn [s] {:foo s})}
+             [:foo :string]]
+            "bar"
+            (mt/transformer
+             (mt/strip-extra-keys-transformer)
+             (mt/string-transformer)))))))
+
+(deftest strip-extra-keys-transformer-test
+
+  (testing "extra keys from :map are stripped"
+    (is (= {:x 1, :y 2}
+           (m/decode
+            [:map [:x :int] [:y :int]]
+            {:x 1, :y 2, :z 3}
+            (mt/strip-extra-keys-transformer)))))
+
+  (testing "extra keys from :map-of are stripped"
+    (is (= {1 1}
+           (m/decode
+            [:map-of :int :int]
+            {1 1, "2" 2, 3 "3", "4" "4"}
+            (mt/strip-extra-keys-transformer))))
+
+    (testing "composing with other transformers"
+      (is (= {1 1, 2 2, 3 3, 4 4}
+             (m/decode
+              [:map-of :int :int]
+              {1 1, "2" 2, 3 "3", "4" "4"}
+              (mt/transformer
                (mt/strip-extra-keys-transformer)
                (mt/string-transformer)))))))
 
-(deftest strip-extra-keys-transformer-test
+  (testing "::m/default defines how extra keys are stripped"
+    (let [value {:x 1, :y 2, :z 3, 1 1, "2" 2, 3 "3", "4" "4"}
+          expected {1 1, :x 1, :y 2}]
+
+      (testing "explicit and default values are both stripped"
```

Comment:
I don't understand this and the previous `testing` line. It looks to me like "explicit and default values are both _kept_"

Reply:
"stripping applies for both explicit and default values"